Adamant/Modules/Settings/Notifications/NotificationsFactory.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  NotificationsFactory.swift
+//  Adamant
+//
+//  Created by Yana Silosieva on 05.08.2024.
+//  Copyright © 2024 Adamant. All rights reserved.
+//
+
+import Swinject
+import SwiftUI
+
+struct NotificationsFactory {
+    private let parent: Assembler
+    private let assemblies = [NotificationsAssembly()]
+    
+    init(parent: Assembler) {
+        self.parent = parent
+    }
+    
+    @MainActor
+    func makeViewController(screensFactory: ScreensFactory) -> UIViewController {
+        let assembler = Assembler(assemblies, parent: parent)
+        let viewModel = { assembler.resolver.resolve(NotificationsViewModel.self)! }
+        
+        let view = NotificationsView(
+            viewModel: viewModel,
+            screensFactory: screensFactory
```

Comment:
The less view knows, the easier it's to test and maintain. I made the universal screensFactory to avoid mass refactoring, but here we are writing the new code.

1. Move the factory of the shown screen out of the ScreensFactory
2. Initialize that factory in this factory.
3. Pass here only the escaping that returns the View instead of the whole factory.

Reference:
_I will insert a link in a few mins_